internal/controller/container_reconciler.go

```diff
@@ -255,3 +290,46 @@ func (r *ContainerImageReconciler) imageName(obj ContainerizedObject) string {
 		panic("unsupported cloud: " + name)
 	}
 }
+
+func (r *ContainerImageReconciler) callSignedUrlGenerator(notebook *apiv1.Notebook, conn *grpc.ClientConn) (string, error) {
+
+	// create the request object
+	req := &sci.CreateSignedURLRequest{
+		BucketName:        r.CloudContext.GCP.ProjectID + "-substratus-notebooks",
+		ObjectName:        "notebook.zip",
+		ExpirationSeconds: 300,
+		Md5Checksum:       notebook.Spec.Image.Upload.Md5Checksum,
+	}
+
+	// Create a client using the connection
+	c := sci.NewControllerClient(conn)
```

Comment:
TODO(bjb): Probably preferable to hoist this to `main` instead of passing down a `*grpc.ClientConn`